website/docs/cdktf/concepts/assets.mdx

```diff
@@ -60,6 +60,160 @@ new MyStack(app, "demo");
+        )
+
+app = App()
+MyStack(app, "demo")
+app.synth()
+```
+
+```java
+import path.*;
+import software.constructs.Construct;
+import com.hashicorp.cdktf.App;
+import com.hashicorp.cdktf.TerraformStack;
+import com.hashicorp.cdktf.TerraformAsset;
+import com.hashicorp.cdktf.AssetType;
+import com.hashicorp.cdktf.providers.aws.AwsProvider;
+import com.hashicorp.cdktf.providers.aws.s3.*;
+
+public class MyStack extends TerraformStack {
+    public MyStack(Construct scope, String name) {
+        super(scope, name);
+
+        AwsProvider.Builder.create(this, "provider")
+                .region("us-west-2")
+                .build();
+
+        S3Bucket bucket = S3Bucket.Builder.create(this, "bucket")
+                .bucket("demo")
+                .build();
+
+        TerraformAsset asset = TerraformAsset.Builder.create(this, "lambda-asset")
+                .path(resolve(__dirname, "../lambda"))
+                .type(AssetType.ARCHIVE)
+                .build();
+
+        S3BucketObject.Builder.create(this, "lambda-archive")
+                .bucket(bucket.getBucket())
+                .key(asset.getFileName())
+                .source(asset.getPath())
+                .build();
+    }
+}
+
+App app = new App();
+new MyStack(app, "demo");
+app.synth();
+```
+
+```csharp
+using Path;
+using Constructs;
+using HashiCorp.Cdktf;
+using HashiCorp.Cdktf.Providers.Aws;
+using HashiCorp.Cdktf.Providers.Aws.s3;
+
+class MyStack : TerraformStack
+{
+    public MyStack(Construct scope, string name) : base(scope, name)
+    {
+
+        new AwsProvider(this, "provider", new AwsProviderConfig {
+            Region = "us-west-2"
+        });
+
+        S3Bucket bucket = new S3Bucket(this, "bucket", new S3BucketConfig {
+            Bucket = "demo"
+        });
+
+        TerraformAsset asset = new TerraformAsset(this, "lambda-asset", new TerraformAssetConfig {
+            Path = Resolve(__dirname, "../lambda"),
+            Type = AssetType.ARCHIVE
+        });
+
+        new S3BucketObject(this, "lambda-archive", new S3BucketObjectConfig {
+            Bucket = bucket.Bucket,
+            Key = asset.FileName,
+            Source = asset.Path
+        });
+    }
+}
+
+App app = new App();
+new MyStack(app, "demo");
+app.Synth();
+```
+
+```go
+import path "github.com/aws-samples/dummy/path"
+import "github.com/aws/constructs-go/constructs"
+import "github.com/hashicorp/terraform-cdk-go/cdktf"
+import "github.com/hashicorp/cdktf-provider-aws-go/aws"
+
+type myStack struct {
+	terraformStack
+}
+
+func newMyStack(scope construct, name *string) *myStack {
+	this := &myStack{}
+	newTerraformStack_Override(this, scope, name)
+
+	aws.NewAwsProvider(this, jsii.String("provider"), &awsProviderConfig{
+		region: jsii.String("us-west-2"),
+	})
+
+	bucket := aws.S3.NewS3Bucket(this, jsii.String("bucket"), &aws.s3.s3BucketConfig{
+		bucket: jsii.String("demo"),
+	})
+
+	asset := cdktf.NewTerraformAsset(this, jsii.String("lambda-asset"), &terraformAssetConfig{
```

Comment:
This also needs to be `&cdktf.TerraformAssetConfig{`

Maybe testing the snippets will find more errors as I'm a bad Go compiler and there could be more missing.